File: src/algorithms/find-disappeared-numbers.js
/**
 * Given an array nums of n integers where nums[i] is in the range [1, n],
 * returns an array of all the integers in the range [1, n] that do not
 * appear in nums.
 * 
 * @param {number[]} nums - An array of numbers
 * @returns {number[]} - Returns an array of numbers
 */
function findDisappearedNumbers (nums) {
  let hash = {};
  for (let i = 0; i < nums.length; i++) {
    hash[nums[i]] = (hash[nums[i]] || 0) + 1;
  }
  let disappearedNums = [];
  for (let i = 1; i < nums.length + 1; i++) {
    if (!hash.hasOwnProperty(i)) {
      disappearedNums.push(i);
    }
  }
  return disappearedNums;
}

File: src/algorithms/find-disappeared-numbers.py
def findDisappearedNumbers(nums):
    """Given an array nums of n integers where nums[i] is in the range [1, n], 
    return an array of all the integers in the range [1, n] that do not appear 
    in nums.

    Args:
        nums (list[int]): A list of integers.

    Returns:
        list[int]: Returns a list of missing integers.
    """
    res = []
    new_set = set(nums)
    for i in range(1, len(nums) + 1):
        if i not in new_set:
            res.append(i)
    return res
